Lib/fontTools/cffLib/cff2_merge_funcs.py

```diff
@@ -0,0 +1,246 @@
+	varTupleIndexes = list(range(len(supports)))
+	varDeltasCFFV = varLib.builder.buildVarData(varTupleIndexes, None, False)
+	varStoreCFFV = varLib.builder.buildVarStore(varTupleList, [varDeltasCFFV])
+
+	topDict = varFont['CFF2'].cff.topDictIndex[0]
+	topDict.VarStore = VarStoreData(otVarStore=varStoreCFFV)
+
+
+def lib_convertCFFToCFF2(cff, otFont):
+	# This assumes a decompiled CFF table.
+	cff2GetGlyphOrder = cff.otFont.getGlyphOrder
+	topDictData = TopDictIndex(None, cff2GetGlyphOrder, None)
+	topDictData.items = cff.topDictIndex.items
+	cff.topDictIndex = topDictData
+	topDict = topDictData[0]
+	if hasattr(topDict, 'Private'):
+		privateDict = topDict.Private
+	else:
+		privateDict = None
+	opOrder = buildOrder(topDictOperators2)
+	topDict.order = opOrder
+	topDict.cff2GetGlyphOrder = cff2GetGlyphOrder
+	if not hasattr(topDict, "FDArray"):
+		fdArray = topDict.FDArray = FDArrayIndex()
+		fdArray.strings = None
+		fdArray.GlobalSubrs = topDict.GlobalSubrs
+		topDict.GlobalSubrs.fdArray = fdArray
+		charStrings = topDict.CharStrings
+		if charStrings.charStringsAreIndexed:
+			charStrings.charStringsIndex.fdArray = fdArray
+		else:
+			charStrings.fdArray = fdArray
+		fontDict = FontDict()
+		fontDict.setCFF2(True)
+		fdArray.append(fontDict)
+		fontDict.Private = privateDict
+		privateOpOrder = buildOrder(privateDictOperators2)
+		for entry in privateDictOperators:
+			key = entry[1]
+			if key not in privateOpOrder:
+				if key in privateDict.rawDict:
+					# print "Removing private dict", key
+					del privateDict.rawDict[key]
+				if hasattr(privateDict, key):
+					delattr(privateDict, key)
+					# print "Removing privateDict attr", key
+	else:
+		# clean up the PrivateDicts in the fdArray
+		fdArray = topDict.FDArray
+		privateOpOrder = buildOrder(privateDictOperators2)
+		for fontDict in fdArray:
+			fontDict.setCFF2(True)
+			for key in fontDict.rawDict.keys():
+				if key not in fontDict.order:
+					del fontDict.rawDict[key]
+					if hasattr(fontDict, key):
+						delattr(fontDict, key)
+
+			privateDict = fontDict.Private
+			for entry in privateDictOperators:
+				key = entry[1]
+				if key not in privateOpOrder:
+					if key in privateDict.rawDict:
+						# print "Removing private dict", key
+						del privateDict.rawDict[key]
+					if hasattr(privateDict, key):
+						delattr(privateDict, key)
+						# print "Removing privateDict attr", key
+	# Now delete up the decrecated topDict operators from CFF 1.0
+	for entry in topDictOperators:
+		key = entry[1]
+		if key not in opOrder:
+			if key in topDict.rawDict:
+				del topDict.rawDict[key]
+			if hasattr(topDict, key):
+				delattr(topDict, key)
+
+	# At this point, the Subrs and Charstrings are all still T2Charstring class
+	# easiest to fix this by compiling, then decompiling again
+	cff.major = 2
+	file = BytesIO()
+	cff.compile(file, otFont, isCFF2=True)
+	file.seek(0)
+	cff.decompile(file, otFont, isCFF2=True)
+
+
+def pointsDiffer(pointList):
```

Comment:
`varLib.models.allEqual()`?

Reply:
I don't see public method varLib.models.allEqual(). I do see varLib.merger._all_equal, which does do the same thing as cffLib.cff_merge_funcs.pointsDiffer, but I like the latter implementation a little better. Not enough to really care, though. Would you prefer I make varLib.merger._all_equal a public method, and use that instead?

Reply:
It should be there now.  Was a branch.  Just merged it.

Your pointsDiffer implementation is multiple times slower than allEqual().